src/main/java/entities/User.java

```diff
@@ -0,0 +1,170 @@
+package java.entities;
+import java.util.ArrayList;
+
+public class User {
+    // ArrayList of User's favourited posts
+    private final ArrayList<Post> favourites;
+
+    // ArrayList of Posts User has made
+    private final ArrayList<Post> posts;
+
+    // boolean relating if User has admin privileges
+    private final boolean isAdmin;
+
+    // Number of Users created
+    public static int numUsers = 0;
+
+    // id is the unique identifier of User, equal to number of posts at time of creation
+    private final int id;
+
+    // password is the user's input password, used to sign in
+    private final String password;
+
+    // email is the users input email, used to sign in
+    private final String email;
+
+    /*
+    * Initializes an instance of User
+    *
+    * @param admin whether the user will have admin priviledges
+    * */
+    public User(boolean admin, int numUsersCreated, String email, String password) {
+        this.favourites = new ArrayList<Post>();
+        this.posts = new ArrayList<Post>();
+        this.isAdmin = admin;
+        User.numUsers = numUsersCreated;
+        User.numUsers ++;
+        this.id = User.numUsers;
+        this.email = email;
+        this.password = password;
+    }
+
+    /*
+    * Returns User's favourites
+    *
+    * @return User's favourites attribute
+    * */
+    public ArrayList getFavourites() {
+        /* Return an ArrayList containing the favourites of this User */
+        return this.favourites;
+    }
+
+    /*
+    * Adds a Post to User's favourites
+    *
+    * @param toAdd The Post to add to User's favourites
+    * */
+    public void addFavourite(Post toAdd) {
+        /* Add Post to this.favourites */
+        this.favourites.add(toAdd);
+    }
+
+    /*
+    * Check if a Post is in User's favourites and remove it if so
+    *
+    * @param toRemove Post to be removed
+    * @return boolean representing success or failure
+    * */
+    public boolean removeFavourite(Post toRemove) {
+        for (int i = 0; i < this.favourites.size(); i++) {
+            if (this.favourites.get(i).equals(toRemove)) {
+                this.favourites.remove(i);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /*
+    * Return Posts made by User
+    *
+    * @return User's posts attribute
+    * */
+    public ArrayList getPosts() {
+        /* return an ArrayList containing the elements of this.posts */
+        return this.posts;
+    }
+
+    /*
+    * Add a Post to User's posts
+    *
+    * @param toAdd post to be added
+    * */
+    public void addPost(Post toAdd) {
+        /* add toAdd to this.posts */
+        this.posts.add(toAdd);
+    }
+
+    /*
+    * Check if a Post is in User's posts and remove it if so
+    *
+    * @param toRemove post to be removed
+    * @returns a boolean representing success or failure
+    * */
+    public boolean removePost(Post toRemove) {
```

Comment:
Maybe rename so we don't mistake it as part of the removePost use case.

Reply:
Is there some convention that helps people label methods so you can easily tell whether they are part of entities or use cases? Otherwise I can't really think of a good way to rename it. Maybe "removePostE"?